### PR TITLE
Delete startupCache if install is newer than it

### PR DESCRIPTION
**Pull request by llewelld**
On installation of embedlite-components-qt5 the package installs the
file /var/lib/_MOZEMBED_CACHE_CLEAN_ .

This change ensures that when the browser is run, the last modified
timestamp of the startupCache is compared against this file, and if the
installed file is newer, the cache is cleared. The startupCache will
then be recreated by xulrunner during execution. Previously the
timestamp was compared against ~/.mozilla/mozembed/_CACHE_CLEAN_ but
since it's no longer createdi it can't be used anymore.

This ensures that any changes made to embedlite javascript components
won't be ignored in favour of older versions in the cache.

### CODE REVIEW

File: src/browser/declarativewebutils.cpp [hunk 1]
```diff
@@ -75,13 +77,14 @@ void DeclarativeWebUtils::clearStartupCacheIfNeeded()
...
-        if (localStamp.exists() && systemStamp.lastModified() > localStamp.lastModified()) {
-            QDir cacheDir(mostProfilePath + "/startupCache");
-            cacheDir.removeRecursively();
-            QFile(localStampString).remove();
+        QString startupCacheDir = QDir::homePath() + gStartupCacheDir;
+        QDir cacheDir(startupCacheDir);
+        if (cacheDir.exists()) {
+            QFileInfoList files = cacheDir.entryInfoList(QDir::Files, QDir::Time);
+            if (files.count() > 0 && systemStamp.lastModified() > files.first().lastModified()) {
+                qDebug() << "Removing cache to synchronise with fresh embedlite-components-qt5";
```
Comment (okodron) [inline src/browser/declarativewebutils.cpp:85]: You can just use printf-like `qDebug("Removing cache to synchronise with fresh embedlite-components-qt5");` as it's quite simple here. Then you don't need to include `<QtCore/QDebug>` as well.

File: src/browser/declarativewebutils.cpp [hunk 2]
```diff
@@ -75,13 +77,14 @@ void DeclarativeWebUtils::clearStartupCacheIfNeeded()
...
     if (systemStamp.exists()) {
-        QString mostProfilePath = QDir::homePath() + gProfilePath;
-        QString localStampString(mostProfilePath + QString("/_CACHE_CLEAN_"));
-        QFileInfo localStamp(localStampString);
-        if (localStamp.exists() && systemStamp.lastModified() > localStamp.lastModified()) {
-            QDir cacheDir(mostProfilePath + "/startupCache");
-            cacheDir.removeRecursively();
-            QFile(localStampString).remove();
+        QString startupCacheDir = QDir::homePath() + gStartupCacheDir;
+        QDir cacheDir(startupCacheDir);
```
Comment (okodron) [inline src/browser/declarativewebutils.cpp:81]: I think it's better to omit intermediate `QString startupCacheDir` variable. Then init `QDir cacheDir(QDir::homePath() + gStartupCacheDir);`.

Comment (llewelld) [reply src/browser/declarativewebutils.cpp]: Thanks, yes, the intermediate string is unnecessary. I've combined the two lines.

File: src/browser/declarativewebutils.cpp (hunk 1 above)
Comment (llewelld) [reply src/browser/declarativewebutils.cpp]: Interesting, I wasn't aware this was possible. I've changed it.

**Comment by llewelld** [comment]
Thank you both for the useful review comments. I'll briefly look into a oneshot before merging.

**Comment by llewelld** [comment]
This PR has been superseded by https://github.com/sailfishos/sailfish-browser/pull/693 and https://git.sailfishos.org/mer-core/gecko-dev/merge_requests/126 (both now merged), so I'm closing this as it's no longer needed.

**Pull request closed without merging**